mrmustard/physics/ansatze.py

```diff
@@ -401,6 +497,372 @@ def __and__(self, other: PolyExpAnsatz) -> PolyExpAnsatz:
         return self.__class__(As, bs, cs)
 
 
+class DiffOpPolyExpAnsatz(PolyExpBase):
+    r"""
+    The ansatz of the Fock-Bargmann representation.
+
+    Represents the ansatz function:
+
+        :math:`F(z) = \sum_i [\sum_k c^(i)_k \partial_y^k \textrm{exp}((z,y)^T A_i (z,y) / 2 + (z,y)^T b_i)|_{y=0}]`
+
+    with ``k`` being a multi-index. The matrices :math:`A_i` and vectors :math:`b_i` are
+    parameters of the exponential terms in the ansatz, and :math:`z` is a vector of variables.
+
+    .. code-block::
+
+        >>> from mrmustard.physics.ansatze import DiffOpPolyExpAnsatz
+
+        >>> A = np.array([[1.0, 0.0], [0.0, 1.0]])
+        >>> b = np.array([1.0, 1.0])
+        >>> c = np.array([1.0,2.0,3.0])
+
+        >>> F = PolyExpAnsatz(A, b, c)
+        >>> z = np.array([1.0, 2.0])
+
+        >>> # calculate the value of the function at ``z``
+        >>> val = F(z)
+
+    Args:
+        A: The list of square matrices :math:`A_i`
+        b: The list of vectors :math:`b_i`
+        c: The array of coefficients for the polynomial terms in the ansatz.
+
+    """
+
+    def __init__(
+        self,
+        A: Optional[Batch[Matrix]] = None,
+        b: Optional[Batch[Vector]] = None,
+        c: Batch[Tensor] = np.array([1.0]),
+        name: str = "",
+    ):
+        self.name = name
+
+        if A is None and b is None:
+            raise ValueError("Please provide either A or b.")
+        A = math.astensor(A)
+        b = math.astensor(b)
+        c = math.astensor(c)
+        super().__init__(mat=A, vec=b, array=c)
+        if self.A.shape[0] != self.c.shape[0] or self.A.shape[0] != self.b.shape[0]:
+            raise ValueError("Batch size of A,b,c must be the same.")
+
+    @property
+    def A(self) -> Batch[ComplexMatrix]:
+        r"""
+        The list of square matrices :math:`A_i`.
+        """
+        return self.mat
+
+    @property
+    def b(self) -> Batch[ComplexVector]:
+        r"""
+        The list of vectors :math:`b_i`.
+        """
+        return self.vec
+
+    @property
+    def c(self) -> Batch[ComplexTensor]:
+        r"""
+        The array of coefficients for the polynomial terms in the ansatz.
+        """
+        return self.array
+
+    def __call__(self, z: Batch[Vector]) -> Scalar:
```

Comment:
I can't really figure out what I'm doing wrong here. I thought I was initializing a Gaussian with two variables, and via c I thought I was telling it to take the 0th derivative on the last variable, then set it to zero (which should just be setting the second variable to zero). In which case, I would expect to just get exp(1/2) if I evaluate the ansatze at 1. But I'm getting that the ansatze evaluates to exp(1).

I have hbar=1 too.
```
A1 = np.eye(2)
b1 = np.zeros((2))
c1 = np.array([1])

ansatze = DiffOpPolyExpAnsatz(A1,b1,c1)
assert ansatze(np.array([1])) == np.exp(1)
assert ansatze(np.array([1])) == np.exp(0.5)
```

Reply:
I got it now. I needed to add a batch dimension to both c and the argument for the call function

Reply:
I updated the call function so you don't need to add a batch to the argument yourself. Also note that the batch on the argument can be different than for the triple. This makes it work equivalent to the old ansatz